### PR TITLE
TRUNK-5060 globalPropertyChanged catches all exceptions #2062

### PR DESCRIPTION
**Pull request by getshravan**
## Description of what I changed
Removed the empty catch block and added a logging when Locale is null

## Checklist: I completed these to help reviewers :)

- [x ] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [x ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x ] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x ] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/main/java/org/openmrs/api/impl/GlobalLocaleList.java [hunk 1]
```diff
@@ -17,58 +17,59 @@
...
-				}
-			}
-			catch (Exception e) {
-				// bad locale spec? just ignore it. the UI should take care of
-				// guiding the user.
+			Locale allowedLocale = LocaleUtility.fromSpecification(allowedLocaleString.trim());
+			if (allowedLocale != null) {
+				allowedLocales.add(allowedLocale);
+			} else {
+				log.error("Attempt to set invalid locale '{}'", allowedLocaleString);
```
Comment (wluyima) [inline api/src/main/java/org/openmrs/api/impl/GlobalLocaleList.java]: Why is this change needed? I thought you just have to remove the try...catch clause and no other change

**Comment by dkayiwa** [comment]
@getshravan are you still working on this?

**Comment by dkayiwa** [comment]
Closing because the author of this pull request no longer works on the ticket.
